internal/internal_nexus_task_poller.go

```diff
@@ -152,7 +152,12 @@ func (ntp *nexusTaskPoller) ProcessTask(task interface{}) error {
 
 	// Execution latency (in-SDK processing time).
 	metricsHandler.Timer(metrics.NexusTaskExecutionLatency).Record(time.Since(executionStartTime))
-	if err != nil || failure != nil {
+
+	// Increment failure in all forms of errors:
+	// Internal error processing the task.
+	// Failure from user handler.
+	// Special case for the start response with operation error.
```

Comment:
Is this a task failure if the process task call doesn't return a failure? I wouldn't expect failure to start things like a child workflow to increment this value.

Reply:
This is when a handler returns an UnsuccessfulOperationError. A user would consider it a failure.

Reply:
I'd be fine renaming the metric to something like `temporal_nexus_operation_execution_failed ` not sure why we need two metrics if activity only needs one

Reply:
My issue with your suggestion is that there are other ways that the execution would fail, e.g. by returning a non retryable error in your handler or the workflow in the workflow run operation fails asynchronously.
For activities we don't distinguish between retryable and non retryable errors, so I figured this model is good enough.
What we have now maps to whether a handler method returned an error or not (or there was an internal error within the SDK).

Reply:
Can you clarify the definition of `temporal_nexus_task_execution_failed`? Is it "any task that fails unexpectedly or sync operation start fails expectedly"? If so, it is confusing to a user that sync start operation RPC returning an operation error causes this counter to increase but async get operation result returning an operation error does not.

Reply:
Definition is, it's incremented if any of the following are true:
1. The user defined handler returns any error (or panics)
2. There's an internal error in the SDK (e.g. bug) while processing the task

Reply:
> The user defined handler returns any error (or panics)

Does this include errors on `GetResult`? Do we really want a metric that can't differentiate between a proper failure and an internal one? Or should we not count operation failures?

Reply:
We haven't implemented `GetResult` but I think we could exclude `ErrOperationStillRunning`. This change includes `UnsuccessfulOperationError` in the metric, where it didn't used to be included before.

The rationale is that all of `HandlerErrorTypeBadRequest`, non-retryable `ApplicationError`, and `UnsuccessfulOperationError` are non-retryable errors from the caller perspective and essentially fail the operation.

Reply:
But I agree that it's not perfect. It wasn't perfect before either.

Reply:
I think the metrics should be split between user-returned operation errors and unexpected failures similar to workflows. People often just stop using activity failure metrics because they are using intentional failures for some things. They have to create their own metrics to differentiate. The difference is more obvious in other languages where unexpected failures/exceptions are more common and the languages are used to differentiating explicit RPC failure from accidental one.